index.js

```diff
@@ -32,10 +39,18 @@ app.get("/", (req, res) => {
     // We're not actually using this in the template yet because it's not random
     let featuredFeeds = []
 
-    const promises = blogRssUrls.map((url) => parser.parseURL(url))
+    const promises = blogUrls.map(async (url) => {
+        feed = await parser.parseURL(blogRegistry[url]["feedLink"]);
```

Comment:
Question for you: doesn't `await` mean that feed actually resolves? I'm assuming this is no longer a promise? If so, how is the `allSettled` call working?

Reply:
An async function (`async () => { ...}`) will always return a Promise object, so even though there's some awaiting inside, the overall return value is asynchronous 😄 

(hopefully that sentence makes sense...)

Reply:
ahhhh so the `async` map is returning those objects (that include the new metadata) as a promise but the value of `feed` is resolved?

Reply:
yes correct! when the function runs, it will wait for the feed to resolve before returning. But the overall function runs asynchronously as a Promise.